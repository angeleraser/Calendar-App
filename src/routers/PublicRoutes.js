import { Redirect, Route } from "react-router-dom";
export const PublicRoute = ({
  path,
  component: Component,
  isAuthenticated,
  ...rest
}) => {
  return (
    <Route
      path={path}
      {...rest}
      component={(props) => {
        return !isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to="/" />
        );
      }}
    />
  );
};
