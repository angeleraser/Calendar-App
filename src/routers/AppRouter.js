import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { BrowserRouter, Redirect, Switch } from "react-router-dom";
import { startChecking } from "../actions/auth";
import { LoginScreen } from "../components/auth/LoginScreen";
import { CalendarScreen } from "../components/calendar/CalendarScreen";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoutes";
export const AppRouter = () => {
  const dispatch = useDispatch();
  const { checking, uid } = useSelector(({ auth }) => auth);
  // Check if token is valid
  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);
  if (checking) {
    return <h1>Loading...</h1>;
  }
  return (
    <BrowserRouter>
      <Switch>
        <PublicRoute
          isAuthenticated={!!uid}
          exact
          path="/login"
          component={LoginScreen}
        />
        <PrivateRoute
          isAuthenticated={!!uid}
          exact
          path="/"
          component={CalendarScreen}
        />
        <Redirect to="/login" />
      </Switch>
    </BrowserRouter>
  );
};
