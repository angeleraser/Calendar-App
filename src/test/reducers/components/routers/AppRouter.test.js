import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { AppRouter } from "../../../../routers/AppRouter";
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
describe("Testing <AppRouter/> component", () => {
  test("should render LoadingScreen", () => {
    const initState = {
      auth: {
        checking: true,
      },
    };
    const store = mockStore(initState);
    const wrapper = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });

  test("should render LoginScreen", () => {
    const initState = {
      auth: {
        checking: false,
      },
    };
    const store = mockStore(initState);
    const wrapper = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
  test("should render CalendarScreen", () => {
    const initState = {
      auth: {
        checking: false,
        name: "Pedroso",
        uid: "asdske2131j4b5k4123hjv42",
      },
      ui: {
        isModalOpen: false,
      },
      calendar: {
        events: [],
        activeEvent: null,
      },
    };
    const store = mockStore(initState);
    const wrapper = render(
      <Provider store={store}>
        <AppRouter />
      </Provider>
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
