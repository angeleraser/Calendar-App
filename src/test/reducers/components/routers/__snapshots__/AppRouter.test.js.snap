// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppRouter/> component should render CalendarScreen 1`] = `
<div>
  <div
    class="calendar-screen"
  >
    <nav
      class="navbar navbar-dark bg-dark mb-4"
    >
      <span
        class="navbar-brand"
      >
        Pedroso
      </span>
      <button
        class="btn btn-outline-danger"
      >
        Salir
      </button>
    </nav>
    <div
      class="rbc-calendar"
    >
      <div
        class="rbc-toolbar"
      >
        <span
          class="rbc-btn-group"
        >
          <button
            type="button"
          >
            Today
          </button>
          <button
            type="button"
          >
            Back
          </button>
          <button
            type="button"
          >
            Next
          </button>
        </span>
        <span
          class="rbc-toolbar-label"
        >
          November 2020
        </span>
        <span
          class="rbc-btn-group"
        >
          <button
            class="rbc-active"
            type="button"
          >
            Month
          </button>
          <button
            class=""
            type="button"
          >
            Week
          </button>
          <button
            class=""
            type="button"
          >
            Day
          </button>
          <button
            class=""
            type="button"
          >
            Agenda
          </button>
        </span>
      </div>
      <div
        class="rbc-month-view"
      >
        <div
          class="rbc-row rbc-month-header"
        >
          <div
            class="rbc-header"
          >
            <span>
              Sun
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Mon
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Tue
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Wed
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Thu
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Fri
            </span>
          </div>
          <div
            class="rbc-header"
          >
            <span>
              Sat
            </span>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg rbc-today"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-now rbc-current"
              >
                <a
                  href="#"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  04
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  05
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  06
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  07
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  08
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  09
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  10
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  11
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  12
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  13
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  14
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  15
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  16
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  17
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  18
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  19
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  20
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  21
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  22
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  23
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  24
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  25
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  26
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  27
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  28
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="rbc-month-row"
        >
          <div
            class="rbc-row-bg"
          >
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
            <div
              class="rbc-day-bg rbc-off-range-bg"
            />
          </div>
          <div
            class="rbc-row-content"
          >
            <div
              class="rbc-row "
            >
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  29
                </a>
              </div>
              <div
                class="rbc-date-cell"
              >
                <a
                  href="#"
                >
                  30
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  01
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  02
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  03
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  04
                </a>
              </div>
              <div
                class="rbc-date-cell rbc-off-range"
              >
                <a
                  href="#"
                >
                  05
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="fab btn btn-primary"
    >
      <i
        class="fa fa-plus"
      />
    </button>
  </div>
</div>
`;

exports[`Testing <AppRouter/> component should render LoadingScreen 1`] = `
<div>
  <h1>
    Loading...
  </h1>
</div>
`;

exports[`Testing <AppRouter/> component should render LoginScreen 1`] = `
<div>
  <div
    class="container login-container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="loginEmail"
              placeholder="Correo"
              type="text"
              value="pedrosete@gmail.com"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="loginPassword"
              placeholder="Contraseña"
              type="password"
              value="123456"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        class="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="registerName"
              placeholder="Nombre"
              type="text"
              value="Angelo"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="registerEmail"
              placeholder="Correo"
              type="email"
              value="angelo123@gmail.com"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="registerPassword1"
              placeholder="Contraseña"
              type="password"
              value="123456"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="form-control"
              name="registerPassword2"
              placeholder="Repita la contraseña"
              type="password"
              value="123456"
            />
          </div>
          <div
            class="form-group"
          >
            <input
              class="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
